src/main/kotlin/org/jitsi/jibri/api/xmpp/XmppApi.kt

```diff
@@ -127,6 +120,7 @@ class XmppApi(
             val environmentContext = createEnvironmentContext(xmppEnvironment, mucClient)
             if (jibriManager.currentEnvironmentContext == environmentContext) {
                 logger.warn("XMPP disconnected, stopping.")
+                statsDClient?.incrementCounter(STOPPED_ON_XMPP_CLOSED, mucClient.tags())
```

Comment:
In the comment for STOPPED_ON_XMPP_CLOSED you mention "A recording session was stopped because XMPP disconnected.". However, right now, this metric only records when Jibri stops, it does not take into account whether a recording is in progress at that time.

Therefore, you should probably first check if there is currentActiveService in jibriManager and only if so increment the STOPPED_ON_XMPP_CLOSED counter.